src/Api/PubnubApi/Security/Crypto/Cryptors/AesCbcCryptor.cs

```diff
@@ -0,0 +1,152 @@
+﻿using PubnubApi.Security.Crypto.Common;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Xml.Linq;
+
+namespace PubnubApi.Security.Crypto.Cryptors
+{
+    public class AesCbcCryptor : CryptorBase
+    {
+        private const string IDENTIFIER = "ACRH";
+
+        public AesCbcCryptor(string cipherKey, IPubnubLog log, IPubnubUnitTest unitTest): base(cipherKey, true, log, unitTest)
```

Comment:
Let's not pass UnitTests in constructors - we can invert it by exposing event delegates

Reply:
the only purpose is pass known IV for encryption, so that we can verify the expected output. Any other option?

Reply:
removed UnitTests from constructors.